src/axom/sidre/core/Group.cpp

```diff
@@ -104,6 +104,136 @@ std::string Group::getPath() const
   return thePath;
 }
 
+/*
+ *************************************************************************
+ *
+ * Return information about data associated with Group subtree with this 
+ * Group at root of tree (default 'recursive' is true), or for this Group 
+ * only ('recursive' is false) in fields of given Conduit Node.
+ *
+ *************************************************************************
+ */
+void Group::getDataInfo(Node& n, bool recursive) const
+{
+  //
+  // Initialize Node fields
+  //
+  IndexType num_groups = 0;
+  IndexType num_views = 0;
+  IndexType num_views_empty = 0;
+  IndexType num_views_buffer = 0;
+  IndexType num_views_external = 0;
+  IndexType num_views_scalar = 0;
+  IndexType num_views_string = 0;
+  IndexType num_bytes_allocated = 0;
+  IndexType num_bytes_external = 0;
+
+  n["num_groups"] = num_groups;
+  n["num_views"] = num_views;
+  n["num_views_empty"] = num_views_empty;
+  n["num_views_buffer"] = num_views_buffer;
+  n["num_views_external"] = num_views_external;
+  n["num_views_scalar"] = num_views_scalar;
+  n["num_views_string"] = num_views_string;
+  n["num_bytes_allocated"] = num_bytes_allocated;
+  n["num_bytes_external"] = num_bytes_external;
+
+  getDataInfoHelper(n, recursive);
+}
+
+/*
+ *************************************************************************
+ *
+ * Private helper method to support getDataInfo() method.
+ *
+ *************************************************************************
+ */
+void Group::getDataInfoHelper(Node& n, bool recursive) const
+{
+  //
+  // Grab Node entries for updating data info for this Group
+  //
+  IndexType num_groups = n["num_groups"].value();
+  IndexType num_views = n["num_views"].value();
+  IndexType num_views_empty = n["num_views_empty"].value();
+  IndexType num_views_buffer = n["num_views_buffer"].value();
+  IndexType num_views_external = n["num_views_external"].value();
+  IndexType num_views_scalar = n["num_views_scalar"].value();
+  IndexType num_views_string = n["num_views_string"].value();
+  IndexType num_bytes_allocated = n["num_bytes_allocated"].value();
+  IndexType num_bytes_external = n["num_bytes_external"].value();
+
+  num_groups += 1;  // count this group
```

Comment:
`getNumGroups()` and `getNumViews()` can be called instead of recomputing those values here.

Reply:
I agree about getNumViews(). Calling getNumGroups() would make things more complicated because this method is recursive.